tracer/src/Datadog.Trace/SpanContextExtractor.cs

```diff
@@ -45,19 +45,35 @@ internal SpanContextExtractor(bool unusedParamNotToUsePublicApi)
             return ExtractInternal(carrier, getter);
         }
 
-        internal static SpanContext? ExtractInternal<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter)
+        /// <inheritdoc />
+        [PublicApi]
+        public ISpanContext? ExtractIncludingDsm<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter, string messageType, string source)
+        {
+            TelemetryFactory.Metrics.Record(PublicApiUsage.SpanContextExtractor_ExtractIncludingDsm);
+            return ExtractInternal(carrier, getter, messageType, source);
+        }
+
+        internal static SpanContext? ExtractInternal<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter, string? messageType = null, string? source = null)
```

Comment:
nit: it's maybe just me, ut  but signatures like this imply that `messageType` and `source` are _both_ optional, whereas in reality if one is supplied, both should be 🤔 In modern .NET you could use a value tuple `(string MessageType, string Source)?` but you'd need to define your own custom `readonly struct` to do that here, so don't know that it's worth it....

Reply:
yes ideally that'd either both be here or both be absent, but given that distinction is already provided by the public methods, so I don't know if it's worth it here.
I could even make that method private since we don't use it ourselves.

Reply:
> I could even make that method private since we don't use it ourselves.

Unfortunately it _is_ used by ourselves, in the auto instrumentation (which you'll need to hook up, see later comment) 😄